app/grandchallenge/reader_studies/views.py

```diff
@@ -1374,89 +1371,85 @@ def get_success_url(self):
         )
 
 
-class DisplaySetInterfacesCreate(ObjectPermissionRequiredMixin, TemplateView):
-    form_class = SingleCIVForm
-    permission_required = (
-        f"{ReaderStudy._meta.app_label}.change_{DisplaySet._meta.model_name}"
-    )
-    raise_exception = True
-    template_name = "reader_studies/display_set_new_interface_create.html"
-
-    def get_permission_object(self):
-        return self.display_set
-
+class DisplaySetInterfacesCreate(InterfacesCreateBaseView):
     @property
-    def display_set(self):
+    def object(self):
         if self.kwargs.get("pk"):
             return DisplaySet.objects.get(pk=self.kwargs["pk"])
-
-    @property
-    def reader_study(self):
-        if self.display_set:
-            return self.display_set.reader_study
         else:
-            return ReaderStudy.objects.get(slug=self.kwargs["slug"])
+            return None
 
-    def get_form_kwargs(self):
-        return {
-            "pk": self.kwargs.get("pk"),
-            "base_obj": self.reader_study,
-            "interface": self.request.GET.get("interface"),
-            "user": self.request.user,
-            "auto_id": f"id-{uuid.uuid4()}",
-            "htmx_url": self.get_htmx_url(),
-        }
+    @property
+    def base_object(self):
+        return ReaderStudy.objects.get(slug=self.kwargs["slug"])
```

Comment:
Shouldn't the base object be `self.object.reader_study`?

Reply:
No, because `self.object` can be `None` here,but we still need the reader study base object in that case

Reply:
Is this bit used for permission checks? If so, you need to be incredibly careful that you do not allow users to be able to modify objects they do not have access to. This often comes from looking up the object via the pk, but then using the URL parameters to get the permission object.